### DISCUSSION TITLE
How to query multiple rows matching two columns, where each value is unique?

### DISCUSSION THREAD

**Discussion by orenlande** [supabase/supabase#17472] [Questions]
I have a pretty simple request to make, and I probably overlooking something, but couldn't figure it out from docs:
Given I have an array of objects:
```
[{cardId: 100, setId: 7}, ..., ...]
```
I need to get the matching row on the relevant table for each object. 

The table on DB is pretty straightforward:
```
| card_id    | set_id    |
| 100        | 7
| 101        | 7
| 200        | 9
```

Tried the following approaches, but none got what I needed:
```
let query = supabase
      .from("cards_data")
      .select("*")

    cards.forEach((card) => {
      query = query
        .eq("card_id", card.cardId)
        .eq("set_id", card.setId);
    });

const { data, error } = await query;
```

```
let query = supabase
      .from("cards_data")
      .select("*")

    cards.forEach((card) => {
     query = query.or(
       `game_card_id.eq.${card.cardId},and(game_set_id.eq.${card.setId})`
      );
    });

const { data, error } = await query;
```

Thanks for any help!

**Comment by GaryAustin1** [comment]
The .or() would be a single filter with lots of pairs of equal tests in it.
Another way would be to concat the two columns in a generated column in the database and do an in test on that with an array of concatenated pairs.
Finally PostgREST (the API used) has the ability to generate a computed field and you could do the .in on that computed field.
https://postgrest.org/en/stable/references/api/computed_fields.html#computed-fields

**Reply by orenlande** [reply]
I think the .or is the most straightforward way to get it done, but it seems I wrote it wrong. The final result returns me way more rows than expected (it should be 30-40 usually, as the deck contains these numbers), but I end up with 300+

Do you have any guidelines on how to write it correctly when using two columns to find a row?

**Reply by GaryAustin1** [reply]
You have to do `.or(and('x.eq.a, y.eq.b), and(x.eq.c,y.eq.d), ....)`
https://postgrest.org/en/stable/references/api/tables_views.html#logical-operators

Remember all of your filter has to fit in the max HTTP URL length, which without checking I think is 2k bytes.
